file1.py

```diff
@@ -0,0 +1,62 @@
+import heapq
+def dijkstra(graph, start, end):
+    distances = {node: float('inf') for node in graph}
+    distances[start] = 0
+    heap = [(0, start)]
+    while heap:
+        (distance, current) = heapq.heappop(heap)
+        if current == end:
+            break
+        for neighbor in graph[current]:
+            new_distance = distance + graph[current][neighbor]
+            if new_distance < distances[neighbor]:
+                distances[neighbor] = new_distance
+                heapq.heappush(heap, (new_distance, neighbor))
+    return distances[end]
+
+
+##############################################################################
+
+def construct_graph(routes, costs):
+    graph = {}
+    for i, route in enumerate(routes):
+        if route[0] not in graph:
+            graph[route[0]] = {}
+        if route[1] not in graph:
+            graph[route[1]] = {}
+        graph[route[0]][route[1]] = costs[i]
+        graph[route[1]][route[0]] = costs[i]
+    return graph
+
+
+###############################################################################
+print("BUS ROUTING SYSTEM")
+print("-"*20)
+print("Enter the number of Routes")
+r=int(input())
+routes=[]
+cost=[]
+for i in range(r):
+    t=[]
+    print("-"*20)
+    print("enter the cites which having routes:-",i+1)
+    print("enter the city from")
+    t.append(input())
+    print("enter the city to")

```

Comment:
Consider taking input in the same line 
eg :  x = input("Enter a value :  ")  , like this.
In this way it's will be btr to understand while executing the prgm.

Reply:
okay